tests/sweep_framework/sweeps/all_gather_T3K_sharded.py

```diff
@@ -0,0 +1,272 @@
+            ttnn.CoreRangeSet({ttnn.CoreRange(ttnn.CoreCoord(0, 0), ttnn.CoreCoord(0, 3))}),
+            ttnn.CoreRangeSet({ttnn.CoreRange(ttnn.CoreCoord(0, 0), ttnn.CoreCoord(7, 0))}),
+            ttnn.CoreRangeSet({ttnn.CoreRange(ttnn.CoreCoord(0, 0), ttnn.CoreCoord(7, 7))}),
+            ttnn.CoreRangeSet({ttnn.CoreRange(ttnn.CoreCoord(0, 0), ttnn.CoreCoord(7, 1))}),
+            ttnn.CoreRangeSet({ttnn.CoreRange(ttnn.CoreCoord(0, 0), ttnn.CoreCoord(10, 7))}),
+            ttnn.CoreRangeSet({ttnn.CoreRange(ttnn.CoreCoord(0, 0), ttnn.CoreCoord(4, 4))}),
+            ttnn.CoreRangeSet({ttnn.CoreRange(ttnn.CoreCoord(0, 0), ttnn.CoreCoord(3, 4))}),
+            ttnn.CoreRangeSet({ttnn.CoreRange(ttnn.CoreCoord(0, 0), ttnn.CoreCoord(11, 7))}),
+        ],
+        "dim": [0, 1, 2, 3],
+        "tensor_layout": [ttnn.ROW_MAJOR_LAYOUT, ttnn.TILE_LAYOUT],
+        "input_dtype": [ttnn.bfloat16, ttnn.bfloat8_b],
+        "orientation": [ttnn.ShardOrientation.ROW_MAJOR, ttnn.ShardOrientation.COL_MAJOR],
+        "tensor_mem_layout": [
+            ttnn.TensorMemoryLayout.WIDTH_SHARDED,
+            ttnn.TensorMemoryLayout.HEIGHT_SHARDED,
+            ttnn.TensorMemoryLayout.BLOCK_SHARDED,
+            ttnn.TensorMemoryLayout.INTERLEAVED,
+            ttnn.TensorMemoryLayout.SINGLE_BANK,
+        ],
+        "all_gather_operation": ["all_gather", "line_all_gather"],
+    },
+}
+
+
+def invalidate_vector(test_vector) -> Tuple[bool, Optional[str]]:
+    # if test_vector["num_links"] == 2:
+    #     return True, f"test cases with num_links = 2 is currently not supported by new mesh fixture"
+    unchunked_input_shape = list(test_vector["input_shape"])
+    unchunked_input_shape[test_vector["dim"]] *= test_vector["num_devices"]
+    if test_vector["num_devices"] < 2:
+        return True, f"Requires multiple devices to run"
+    elif test_vector["num_devices"] == 2 and test_vector["num_links"] == 2:
+        return True, f"Not enough links to run"
+
+    if unchunked_input_shape[test_vector["dim"]] % test_vector["num_devices"] != 0 or (
+        test_vector["dim"] == 3 and unchunked_input_shape[test_vector["dim"]] // test_vector["num_devices"] % 32 != 0
+    ):
+        return True, f"Unsupported test case"
+    if test_vector["tensor_layout"] == ttnn.ROW_MAJOR_LAYOUT and test_vector["input_dtype"] == ttnn.bfloat8_b:
+        return True, f"bfloat8_b/4_b only supports TILE layout"
+    if test_vector["tensor_layout"] == ttnn.ROW_MAJOR_LAYOUT:
+        return True, f"ROW_MAJOR_LAYOUT not supported"
+    if test_vector["tensor_mem_layout"] == ttnn.TensorMemoryLayout.WIDTH_SHARDED:
+        return True, f"Output mismatch"
+    if test_vector["num_links"] == 2 and test_vector["num_devices"] == 8:
+        return True, f"8 devices and 2 links are not supported"
+    if test_vector["input_shard_shape"][1] != test_vector["input_shape"][-1]:
+        return True, f"Shard width must be equal to last dim of shape"
```

Comment:
I don't think this is true. For aligned shards it'll need to factor in shard grid and it'll depend on the shard strategy. Delete this.

Width sharded - width dim matches shard size width  *  shard grid num cores.
Height is similar except you use shard height instead of width.
Block would need both dims checked independently.

The above are only checked for unpadded sharded tensors